text/0000-naming-conventions.md

```diff
@@ -0,0 +1,267 @@
+- Start Date: 2018-03-21
+- RFC PR: (leave this empty)
+- WordPress Coding Standards Issue: (leave this empty)
+
+# Summary
+
+This guide is intended to maximize consistency when writing code in the WordPress environment.
+Naming your language constructs is important and should reflect the nature of what it does.
+
+# Basic example
+
+Naming variables, functions, classes, namespaces should be unified across the core, themes and plugins.
+
+Variables should be clear and understandable, written in lowercase using hyphens as a separator.
+
+```php
+$post_array;
+```
+
+Classes (namespaces, traits, interfaces, abstract classes and classes) should be written in capitalized words separated by underscores.
```

Comment:
There's no distinction here between `abstract classes` and `classes` (just as with `final classes`, for example).